lib/Runtime/PlatformAgnostic/Platform/Unix/DateTime.cpp

```diff
@@ -0,0 +1,103 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "Common.h"
+#include "ChakraPlatform.h"
+#include <CoreFoundation/CFDate.h>
+#include <CoreFoundation/CFTimeZone.h>
+
+namespace PlatformAgnostic
+{
+namespace DateTime
+{
+    static void CopyTimeZoneName(WCHAR *wstr, size_t *length, const char *tm_zone)
+    {
+        *length = strlen(tm_zone);
+
+        for(int i = 0; i < *length; i++)
+        {
+            wstr[i] = (WCHAR)tm_zone[i];
+        }
+
+        wstr[*length] = (WCHAR)0;
+    }
+
+    const WCHAR *Utility::GetStandardName(size_t *nameLength, const DateTime::YMD *ymd)
+    {
+        AssertMsg(ymd != NULL, "xplat needs DateTime::YMD is defined for this call");
+        double tv = Js::DateUtilities::TvFromDate(ymd->year, ymd->mon, ymd->mday, ymd->time);
+        int64_t absoluteTime = tv / 1000;
+        absoluteTime -= kCFAbsoluteTimeIntervalSince1970;
+
+        CFTimeZoneRef timeZone = CFTimeZoneCopySystem();
+
+        int offset = (int)CFTimeZoneGetSecondsFromGMT(timeZone, (CFAbsoluteTime)absoluteTime);
+        absoluteTime -= offset;
+
+        char tz_name[128];
+        CFStringRef abbr = CFTimeZoneCopyAbbreviation(timeZone, absoluteTime);
+        CFStringGetCString(abbr, tz_name, sizeof(tz_name), kCFStringEncodingASCII);
```

Comment:
Our xplat expectation for tz_name is non-wide char from the source. Your suggestion may also work indeed.